shared/util.py

```diff
@@ -347,26 +347,39 @@ def get_conversations(user_id):
         db_client = CosmosClient(AZURE_DB_URI, credential, consistency_level='Session')
         db = db_client.get_database_client(database=AZURE_DB_NAME)
         container = db.get_container_client('conversations')
-        conversations = container.query_items(query='SELECT * FROM c WHERE c.conversation_data.interactions[0].user_id  = @user_id', parameters=[dict(name='@user_id', value=user_id)], enable_cross_partition_query=True)
-        formatted_conversations = [ {'id': con['id'], 'start_date': con['conversation_data']['start_date'], 'content': con['history'][0]['content']} for con in conversations]
+        
+        query = 'SELECT * FROM c WHERE c.conversation_data.interaction.user_id = @user_id'
+        parameters = [dict(name='@user_id', value=user_id)]
+        
+        conversations = container.query_items(query=query, parameters=parameters, enable_cross_partition_query=True)
+        
+        formatted_conversations = [{'id': con['id'], 'start_date': con['conversation_data']['start_date'], 'content': con['conversation_data']['history'][1]['content']} for con in conversations]
+        
         return formatted_conversations
     except Exception:
         logging.error("Error retrieving the conversations")
         return []
 
+
+
 def get_conversation(conversation_id, user_id):
     try: 
         credential = DefaultAzureCredential()
         db_client = CosmosClient(AZURE_DB_URI, credential, consistency_level='Session')
         db = db_client.get_database_client(database=AZURE_DB_NAME)
         container = db.get_container_client('conversations')
         conversation = container.read_item(item=conversation_id, partition_key=conversation_id)
-        if conversation['conversation_data']['interactions'][0]['user_id'] != user_id:
-            return {}
-        return conversation
+        if conversation['conversation_data']['interaction']['user_id'] != user_id:
+            return []
+        formatted_conversation = [{
+            'role': message['role'],
+            'content': message['content']
+        } for message in conversation['conversation_data']['history'][1:]]
+        return formatted_conversation
     except Exception:
         logging.error(f"Error retrieving the conversation '{conversation_id}'")
-        return {}
+        return []
```

Comment:
@KenyerRamirez is this suppose to return a single conversation or a set of conversations ?